apps/sms/views/shared/js/utils.js

```diff
@@ -711,6 +711,57 @@
     },
 
     /**
+     * Returns a rate limited function, useful to reduce the number of function
+     * calls when listening to user input (keypress) or browser events.
+     * @params {function} func - The function that should be rate-limited
+     * @params {number=300} delay - Min amount of time (ms) between two calls
+     * @param {Boolean=false} preventfirstCall - Prevent first synchronous call 
+     * when true.
+     * @param {Boolean=false} preventLastCall - Prevent last trailing call
+     * when true.
+     * @returns {function} - The rate limited function that will call the
+     *  original function
+     */
+    throttle: function(func,
+                       delay = 300, 
+                       preventfirstCall = false, 
```

Comment:
I was thinking that maybe you can set delay = 300 and preventfirstCall = true to default here and then you don't have to add 300/true as parameter in conversation.js. But if you still want to specify the delay and preventfirstCall in conversations, maybe you can make the delay/preventfirstCall/preventLastCall into an option for readability:
```
const THROTTLE_OPTS = {
  delay: 300,
  preventfirstCall : true,
  ...
}
``` 
And use this default option instead, wdyt?